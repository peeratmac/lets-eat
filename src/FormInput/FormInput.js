import React, { Component } from 'react';

class FormInput extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      date: '',
      time: '',
      number: null
    };
  }

  handleChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  };

  render() {
    return (
      <div>
        <input
          onChange={this.handleChange}
          type='text'
          name='name'
          value={this.state.name}
          placeholder='Name'
        />
        <input
          onChange={this.handleChange}
          type='text'
          name='date'
          value={this.state.date}
          placeholder='Date (MM/DD)'
        />
        <input
          onChange={this.handleChange}
          type='text'
          name='time'
          value={this.state.time}
          placeholder='Time (X:XX am/pm)'
        />
        <input
          onChange={this.handleChange}
          type='text'
          name='number'
          value={this.state.number}
          placeholder='Number of Guests'
        />
        <button>Make Reservation</button>
      </div>
    );
  }
}

export default FormInput;
