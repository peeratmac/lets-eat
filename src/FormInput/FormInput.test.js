import React from 'react';
import { shallow } from 'enzyme';
import FormInput from './FormInput';

describe('FORM INPUT', () => {
  it('should match the FormInput Component Snapshot', () => {
    const mockMakeReservation = jest.fn();
    const wrapper = shallow(
      <FormInput makeReservation={mockMakeReservation} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
