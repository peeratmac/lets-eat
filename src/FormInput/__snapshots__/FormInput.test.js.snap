// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FORM INPUT should match the FormInput Component Snapshot 1`] = `
<div>
  <input
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    name="date"
    onChange={[Function]}
    placeholder="Date (MM/DD)"
    type="text"
    value=""
  />
  <input
    name="time"
    onChange={[Function]}
    placeholder="Time (X:XX am/pm)"
    type="text"
    value=""
  />
  <input
    name="number"
    onChange={[Function]}
    placeholder="Number of Guests"
    type="text"
    value=""
  />
  <button
    className="make-reservation-button"
    onClick={[Function]}
  >
    Make Reservation
  </button>
</div>
`;
